nifi-commons/nifi-web-client-api/src/main/java/org/apache/nifi/web/client/api/HttpRequestBodySpec.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.web.client.api;
+
+import java.io.InputStream;
+import java.util.OptionalLong;
+
+/**
+ * HTTP Request Body Specification builder
+ */
+public interface HttpRequestBodySpec extends HttpRequestHeadersSpec {
+    /**
+     * Set Request Body as stream
+     *
+     * @param inputStream Request Body stream is required
+     * @param contentLength Content Length or empty when not known
+     * @return HTTP Request Headers Specification builder
+     */
+    HttpRequestHeadersSpec body(InputStream inputStream, OptionalLong contentLength);
```

Comment:
What do you think of overloading this method with different parameters? In order to avoid typing OptionalLong.empty most of the time.
- HttpRequestHeadersSpec body(InputStream inputStream) and
- HttpRequestHeadersSpec body(InputStream inputStream, Long contentLength);

Reply:
Thanks for the suggestion. For the moment, it seems better to avoid the convenience method since the request body size should be available in many cases. Although the request content length header is not necessarily required, it is useful when known, so having the single method provides a prompt to consider whether it is available. I am open to revisiting this and other features as it moves and forward and gets integrated.